base.cmake

```diff
@@ -177,8 +208,79 @@ MACRO(SETUP_PROJECT)
   _SETUP_PROJECT_UNINSTALL()
   _SETUP_PROJECT_PKG_CONFIG()
   _SETUP_PROJECT_DOCUMENTATION()
+  _SETUP_PROJECT_PACKAGE_INIT()
 ENDMACRO(SETUP_PROJECT)
 
+# _SETUP_PROJECT_PACKAGE_FINALIZE
+# -------------
+#
+# Initialize the PackageConfig installation configuration.
+# https://github.com/forexample/package-example
+#
+# This function does not take any argument but check that some
+# variables are defined (see documentation at the beginning of this
+# file).
+#
+# assumes SETUP_PROJECT() was called
+# internally the real requirement is that _SETUP_PROJECT_PACKAGE_INIT() was called
+MACRO(_SETUP_PROJECT_PACKAGE_FINALIZE)
+
+####
+# Installation (https://github.com/forexample/package-example)
+
+# Layout. This works for all platforms:
+#   * <prefix>/lib/cmake/<PROJECT-NAME>
+#   * <prefix>/lib/
+#   * <prefix>/include/
+set(CONFIG_INSTALL_DIR "lib/cmake/${PROJECT_NAME}")
+set(INCLUDE_INSTALL_DIR "include")
+set(INCLUDE_INSTALL_DESTINATION "${INCLUDE_INSTALL_DIR}/${PROJECT_NAME}")
+
+set(GENERATED_DIR "${CMAKE_CURRENT_BINARY_DIR}/generated")
+
+# Configuration
+set(VERSION_CONFIG "${GENERATED_DIR}/${PROJECT_NAME}ConfigVersion.cmake")
+set(PROJECT_CONFIG "${GENERATED_DIR}/${PROJECT_NAME}Config.cmake")
+set(TARGETS_EXPORT_NAME "${PROJECT_NAME}Targets")
+set(namespace "${PROJECT_NAME}::")
+
+# Include module with fuction 'write_basic_package_version_file'
+include(CMakePackageConfigHelpers)
+
+# Configure '<PROJECT-NAME>ConfigVersion.cmake'
+# Note: PROJECT_VERSION is used as a VERSION
+write_basic_package_version_file(
+    "${VERSION_CONFIG}" COMPATIBILITY SameMajorVersion
```

Comment:
`PROJECT_VERSION` is always set by jrl-cmakemodules regardless of the users' inputs (at worst it will be `UNKNOWN` if there is no version provided otherwise and no git tag to find the version)